newrelic-agent/src/test/java/com/newrelic/agent/logging/ClearExpiredLogFilesRunnableTest.java

```diff
@@ -0,0 +1,178 @@
+package com.newrelic.agent.logging;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class ClearExpiredLogFilesRunnableTest {
+
+    private ClearExpiredLogFilesRunnable clearExpiredLogFilesRunnable;
+    private Path tempDir;
+
+    @Before
+    public void setUp() throws IOException {
+        String fileNamePrefix = "testLogDir";
+        tempDir = Files.createTempDirectory(fileNamePrefix);
+    }
+
+    @After
+    public void tearDown() throws IOException {
+        if (tempDir != null && Files.exists(tempDir)) {
+            Files.walk(tempDir)
+                    .sorted((a, b) -> -a.compareTo(b))
```

Comment:
Do they need to be sorted?

Reply:
The only test that requires files to be sorted prior to deletion is `testDirectoryWithMixedFiles`. This test contains a subdirectory with files so the sorting ensures that all files are deleted before subdirectories are deleted. Without sorting, this test throws a `DirectoryNotEmptyException` when attempting to delete the parent directory and the test fails.